messaging/src/main/java/org/axonframework/usage/detection/TestEnvironmentDetector.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.usage.detection;
+
+import java.util.Arrays;
+
+/**
+ * This class detects whether the usage reporter is being used in a unit test environment, such
+ * as a JUnit test or another testing framework.
+ *
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class TestEnvironmentDetector {
+
+    /**
+     * Checks whether the current environment is a test environment.
+     *
+     * @return {@code true} if the current environment is a test environment, {@code false} otherwise
+     */
+    public static boolean isTestEnvironment() {
+        if(System.getProperty("axoniq.usage.force-test-environment", "false").equals("true")) {
+            return false; // Skip detection if explicitly configured
+        }
+        return Arrays.stream(new Throwable().getStackTrace())
```

Comment:
`Thread.currentThread().getStackTrace()` might be slightly cleaner to obtain a stacktrace.